### PR TITLE
Add V0 No-Drop Nuts

### PR DESCRIPTION
**Pull request by zruncho3d**
Working nicely.

  * [x] The mod, firmware configuration or slicer profile is in the correct category
    folder. Printable mods go to `printer_mods/`, firmware configurations
    go to `firmware_configurations/`, slicer profiles go to `slicer_profiles/`.
    Create a subfolder with your name, and place the mods in a subfolder with
    a descriptive name within that folder, e.g.: `/printer_mods/FHeilmann/flux_capacitor`
  * [x] Folders names MUST NOT contain spaces. If possible, make sure file names also 
    do not contain any spaces.
  * [x] For each mod, add a small `README.md` file to its folder with a short description
    of what the mod accomplishes. This readme can be used to add pictures, give assembly
    instructions or specify a bill of materials if the mod requires additional hardware.
  * [x] The PR modifies the top-level `README.md` of the category folder adding the 
    contribution to the table. Read the top part of the file for instructions on how
    to do this. Please preserve the alphabetical ordering while adding new rows. Make sure
    to fill out the compatibility matrix to indicate which versions of the Voron printer
    the submission is compatible with.
  * [x] The mod/configuration/profile has been tested by the person submitting the mod 
    and/or other Voron users. Make sure to add information about how the mod was tested below. 
  * [x] The mod is not merely a slight modification of an official Voron part, configuration 
    or profile (i.e. an official Voron part with a few mm added or removed or a slicer profile 
    which only modifies a few values). *(When in doubt, contact one of the admins in the 
    Voron discord before submitting the PR)*
  * [x] Submitted STLs are printable without support. *(If the mod does not meet this criterion
    join the Voron discord and ask the other users for advice on how to modify the mod such 
    that it does not require supports)*
  * [x] Submitted STL files are not corrupt. *(This can be tested by opening the STL in PrusaSlicer
    and checking if mesh errors are reported.)*
  * [x] Submitted STL files are oriented and scaled properly for printing.
  * [x] Submitted firmware configs or slicer profiles contain no sensitive data (e.g. API keys).

#### Which mods/configurations/profiles are added by this PR?

Adds a family of LDO and MakerBeamXL-compatible nut holders that make improve the V0 assembly and disassembly experience.  

I say this as someone who build a V0 and found the experience much harder than it should be, and wanted to improve it for  a second V0.

#### How was it tested? 

I build a V0 assembly with these.  This came _after_  ~15x failed iterations.

Grantr printed ~60 of the 15mm and found them useful on a different V0 build.

#### Any background context you want to provide?

Even if there are other similar ones, this PR is different because there's a family of sizes and they provide alignment as well.

#### Screenshots (if appropriate)

See commmit README.

### CODE REVIEW

**Review by KiloQubit — COMMENTED** [review]
There's already a mod for V0 nuts that don't slide around here https://github.com/VoronDesign/VoronUsers/tree/master/printer_mods/Zen3D/V0_1515_T-nut .  We try to avoid duplicate mods.  Are there some advantages to your mod over the existing one?

**Comment by zruncho3d** [comment]
Yes, there are multiple advantages, as noted in the commit message above, and the README.

There's a family of sizes, and by having multiple sizes and shapes, you get automatic alignment as well.  That speeds up the Z assembly build.  Plus, with a 15mm-width version, you can align them for the AB blocks without thought, just by pressing them against each other.  And then with double-nut versions, you get easier alignment for the main Z nut, as well as the back-side nuts on the Z assembly.  

It's faster, and easier, to align to the edge of an extrusion, vs having to put nuts in and then move them around.  The FIRST time you put the Z assembly No-Drop Nuts in, you are guaranteed alignment.

**Comment by Zen3Dlab** [comment]
Does it help if I rename my mod to "Simple V0 1515 T-Nut"?

**Comment by zruncho3d** [comment]
I effectively built another V0 today, mostly with the NoDropNuts.  I printed a second batch with infill and the nuts were too tight in the extrusions, so lesson learned there: read your own instructions :-)   Yes, they really helped out - especially for the enclosure, mid-panel, and skirts.

The only issue I saw was that for some parts which already have a boss to index into an extrusion (Nevermore Micro mounts), the little printed thumb grips get in the way, but I found that with pliers I could remove those.  A version without the thumb grips for exactly these cases would make sense, though none of the standard V0 parts will have this issue.

File: printer_mods/zruncho/V0_No_Drop_Nuts/README.md
Comment (KiloQubit) [inline printer_mods/zruncho/V0_No_Drop_Nuts/README.md:41]: This sentence is incomplete, `plenty for the` what?

File: printer_mods/zruncho/V0_No_Drop_Nuts/README.md
Comment (KiloQubit) [inline printer_mods/zruncho/V0_No_Drop_Nuts/README.md:44]: Can you add an explanation of which filenames go where?  It's not clear where `NoDropNut_Rear_Ext_Rear_Edge_x2.stl` is supposed to go for example

**Review by KiloQubit — CHANGES_REQUESTED** [review]
Thanks for the explanation!

I just have some comments on improving the README below.

**Comment by Fulg** [comment]
It took me way too long to realize those new nut sleeves do not fit the BOM extrusions (Makerbeam XL). So this is one important downside of this new (albeit improved) version.

File: printer_mods/zruncho/V0_No_Drop_Nuts/README.md
Comment (zruncho3d) [reply printer_mods/zruncho/V0_No_Drop_Nuts/README.md:41]: Just pushed an update that should it abundantly clear what goes where.  There already was a pic, but now there's a list to help.  Also fixed the incomplete sentence typo.

Also, I clarified even more that LDO is currently the only supported extrusion type.  

And I've ordered some MakerBeamXL to test, to add to this mod, but there's no reason to wait for that.

@KiloQubit - anything else here you'd like before this can be merged?  Thanks.

**Comment by zruncho3d** [comment]
@Fulg I've made that more clear in the title.. but yes, I'd like to add Makerbeam XL to the supported types list.  Ideally with one file type, and I've ordered some MakerBeamXL to test.  

🤞 that this is good to go, as is!

File: printer_mods/README.md
Comment (KiloQubit) [inline printer_mods/README.md:163]: The `for` doesn't make sense in the description there.  And change `LDO only` to `LDO frame only` or `LDO extrusions only`.

File: printer_mods/zruncho/V0_No_Drop_Nuts/README.md
Comment (KiloQubit) [inline printer_mods/zruncho/V0_No_Drop_Nuts/README.md:1]: Change `LDO only` to `LDO frame only` or `LDO extrusions only` to avoid confusion, since LDO makes many products.

**Review by KiloQubit — CHANGES_REQUESTED** [review]
Thanks for the update!  Just a few more minor fixes.

File: printer_mods/zruncho/V0_No_Drop_Nuts/README.md
Comment (zruncho3d) [reply printer_mods/zruncho/V0_No_Drop_Nuts/README.md]: Fixed

**Comment by zruncho3d** [comment]
@KiloQubit  I've address the remaining minor items, and I see that there's a conflict here.  In general, for this repo, do you prefer authors to rebase their commits to fix conflicts, or do maintainers resolve conflicts at merge time?  

I'm used to more of a gerrit worfklow than a GitHub one.  Thanks.

**Comment by KiloQubit** [comment]
Generally the maintainers resolve conflicts, because most people submitting PRs to this repo don't know how to do that.  I've resolved the one in this PR.

However, it's definitely helpful for the author to resolve the conflicts if they know how to.  Rebasing is fine if the PR has not been reviewed yet.  However, after someone reviewed a PR, it's preferable to do a merge to resolve conflicts, rather than rebase, since a rebase makes it more difficult to see what has changed since the review.  All commits are squashed on the final merge anyway.

**Review by KiloQubit — APPROVED** [review]
Thanks for the update!

**Comment by zruncho3d** [comment]
@KiloQubit Got it on rebases; given the structure of this repo and relative independence of contributions to it, I would have to assume that 99-100% of conflicts would be in READMEs anyway, and so having maintainers do merge-conflict resolution would be reasonable.    

The difficulty of reasoning about changes in a PR after a rebase was exactly why I asked before acting here; thanks for clarifying that, as well as the squash-on-merge policy.  Expect to see some more PRs coming your way soon enough!

**Comment by Fulg** [comment]
I don't know if I should log a separate issue but this seems like writing it here will reach the appropriate audience...

It seems the spacing for the leadscrew nut got wider in V0.1, so the nut capture that goes in the back of the bed mount cannot be used. Unfortunately you will only find this out much later in your V0.1 build, when you try to mount the leadscrew nut, and the bed assembly is long done...

Thankfully if you used the No-drop nuts everywhere for your bed assembly, you can dismantle the left side to access one side of the back extrusion and exchange the middle part for 2 separate ones, then put back the screws as if nothing happened _because none of the nuts moved_.

**Pull request merged**
